pages/instructorFileBrowser/instructorFileBrowser.js

```diff
@@ -213,7 +194,7 @@ function browseDirectory(file_browser, callback) {
             fs.lstat(filepath, (err, stats) => {
               if (ERR(err, callback)) return;
               if (stats.isFile()) {
-                const editable = canEditFile(filepath);
+                const editable = modelist.getModeForPath(filename)?.extRe?.test(filename);
```

Comment:
I was just looking at Matt's comment over on https://github.com/PrairieLearn/PrairieLearn/issues/1985 - is there any reason not to use `isBinary` here? IMO we should allow folks to edit any text files, even if we don't have syntax highlighting for them.

Reply:
What comes to mind is that `isBinary` requires this operation to check the content of every file in the directory. I don't mind doing that in line 323, where we read the contents anyway, but in here it would require additional I/O. Do you think that's worth it?

Reply:
Changed to `isText` for file browsing, let me know your thoughts on doing this for directory browsing as well.

Reply:
Ah, great point about reading every file... I suppose one option is to do what GitHub does and only show the action buttons (Edit, etc.) on the "view individual file" page? Having the "Edit" button behave differently depending on which page it's on is kind of strange to me.

Reply:
I don't like requiring two clicks instead of one for operations that may be relatively common for instructors using the Web interface. I get your concern about consistency, not sure what the best approach is in that case.

Reply:
@jonatanschroeder check out the change for `isBinaryFile` in https://github.com/PrairieLearn/PrairieLearn/pull/6370/files: instead of reading all the contents into memory, you can give the function a path and it will read just the first few thousand bytes. It's definitely still *more* I/O than none at all, but it should still be plenty fast for an average directory with no more than a few dozen files.

Reply:
Testing this change I found this: https://github.com/PrairieLearn/PrairieLearn/pull/6370/files#r981661573. I can fix it in this PR, just let me know your thoughts.